swift/extractor/visitors/ExprVisitor.h

```diff
@@ -84,6 +84,11 @@ class ExprVisitor : public AstVisitorBase<ExprVisitor> {
   codeql::BridgeFromObjCExpr translateBridgeFromObjCExpr(const swift::BridgeFromObjCExpr& expr);
   codeql::DotSelfExpr translateDotSelfExpr(const swift::DotSelfExpr& expr);
   codeql::ErrorExpr translateErrorExpr(const swift::ErrorExpr& expr);
+  // following requires non-const because:
+  // * `swift::UnresolvedPatternExpr::getSubPattern` gives `const swift::Pattern*` on const refs
```

Comment:
I think this should be:
```suggestion
  // * `swift::UnresolvedPatternExpr::getSubPattern` gives `const swift::Pattern*` no const refs
```
or I simply don't understand what you're trying to say here.

Reply:
I'll reword, it is confusing.

What I wanted to say, is that this is one rare cases where the swift frontend library went for const correctness and provided a const qualified method, so `getSubPattern` gives back `const swift::Pattern*` on `const UnresolvedPatternExpr&` (practically all other getters in the library do not have a `const` overload). It would be good, if they also provided a version of the visitors working on `const` pointers, but they don't.